### PR TITLE
feat: docs for subscriptions, subscriptions + auth, subscriptions + amq

### PR DESCRIPTION
**Pull request by darahayes**
## Motivation

https://issues.jboss.org/browse/AEROGEAR-9138
https://issues.jboss.org/browse/AEROGEAR-9139

# What

1. Replaces a small amount of documentation on configuring real time updates on the server side using `subscriptions-transport-ws` with new documentation on configuring real time updates using `aerogear/voyager-subscriptions.` [Preview](https://deploy-preview-714--aerogear-docs-preview.netlify.com/aerogear/latest/ds-realtime/#realtime-updates-{context})

2. Adds documentation around using @aerogear/graphql-mqtt-subscriptions for pubsub in the data sync server [Preview](https://deploy-preview-714--aerogear-docs-preview.netlify.com/aerogear/latest/ds-realtime/#mqtt-pubsub-{context})

3. Documentation around implementing Authentication and Authorization over websockets in the data sync server. [Preview](https://deploy-preview-714--aerogear-docs-preview.netlify.com/aerogear/latest/ds-auth/#authentication-and-authorization-websockets-{context})

4. Documentation around implementing using AMQ Online for Pubsub in the data sync server. [Preview](https://deploy-preview-714--aerogear-docs-preview.netlify.com/aerogear/latest/ds-realtime/#configuring-amq-online-for-mqtt-messaging)

### CODE REVIEW

File: modules/ROOT/pages/_partials/data-sync/server-realtime-updates-amq.adoc [hunk 1]
```diff
@@ -0,0 +1,283 @@
...
+      operations: ["send", "recv"]
+----
+
+Create the messaging user.
+
+----
+oc create -f my-messaging-user.yaml
+----
+
+An application can now connect to an AMQ Online address using this ``MessagingUser``'s credentials. For more information see https://access.redhat.com/documentation/en-us/red_hat_amq/7.2/html-single/using_amq_online_on_openshift_container_platform/index#con-user-model-messaging[AMQ Online User Model].
```
Comment (wtrocki) [inline modules/ROOT/pages/_partials/data-sync/server-realtime-updates-amq.adoc]: We do not use any styling above
```suggestion
An application can now connect to an AMQ Online address using this MessagingUser's credentials. For more information see https://access.redhat.com/documentation/en-us/red_hat_amq/7.2/html-single/using_amq_online_on_openshift_container_platform/index#con-user-model-messaging[AMQ Online User Model].
```

File: modules/ROOT/pages/_partials/data-sync/server-realtime-updates.adoc [hunk 2]
```diff
@@ -2,83 +2,117 @@
...
+
+  createSubscriptionServer({ schema: apolloServer.schema }, {
+    server,
+    path: '/graphql'
+  })
+)
+----
++
+`createSubscriptionServer` installs handlers for managing websocket connections and delivering subscriptions on the server. It is a thin wrapper around `SubscriptionServer` from the popular https://npm.im/subscriptions-transport-ws[subscriptions-transport-ws] module that provides integrations with other modules such as `@aerogear/voyager-keycloak`.
+`createSubscriptionServer` returns a `SubscriptionServer` instance and it supports all of the same arguments and options.
```
Comment (finp) [inline modules/ROOT/pages/_partials/data-sync/server-realtime-updates.adoc]: same options as https://npm.im/subscriptions-transport-ws ?

Comment (darahayes) [reply modules/ROOT/pages/_partials/data-sync/server-realtime-updates.adoc]: Kind of but not exactly. The same options as the `SubscriptionServer` class from the `subscriptions-transport-ws` module (which has a couple of classes).

**Comment by darahayes** [comment]
Woo! Happy Friday!

**Pull request merged**